state/protocol/kvstore_test.go

```diff
@@ -0,0 +1,49 @@
+package protocol
+
+import (
+	"math/rand"
+	"testing"
+
+	clone "github.com/huandu/go-clone/generic"
+	"github.com/onflow/go-ethereum/rlp"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+// TestExecutionMeteringParameters_EncodeRLP tests properties of the custom RLP encoding logic.
+func TestExecutionMeteringParameters_EncodeRLP(t *testing.T) {
+	params1 := &ExecutionMeteringParameters{
+		ExecutionMemoryLimit: rand.Uint64(),
+		ExecutionMemoryWeights: map[uint32]uint64{
```

Comment:
Even if the implementation is incorrect the test would pass in 17% of cases. That seems a bit high. Can we fill in some more values to reduce this closer to 0.

Reply:
Good point. I think it is actually 2.7% because you need the right permutation of both 3-item maps: $\frac{1}{3!} \frac{1}{3!} = 0.027...$ Still on the high side though.  https://github.com/onflow/flow-go/pull/7020/commits/a1498df3493b62a6a644ba94d0a7d9ba12b710ac